src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -405,14 +404,19 @@ async def run_setup_pys(
         )
         exported_targets = list(FrozenOrderedSet(owners))
 
-    py2 = is_python2(
-        python_setup.compatibilities_or_constraints(
-            target_with_origin.target.get(PythonInterpreterCompatibility).value
+    interpreter_constraints = PexInterpreterConstraints.create_from_compatibility_fields(
```

Comment:
This refactor was convenient because we added similar new methods to `PexInterpreterConstraints` for Black and MyPy. Now they're located together.